File: euler/other_imp/dijkstra.c
#include <stdio.h>
#include <stdlib.h>
#define infinity 9999

int main()
{
	int n,v,i,j,u,w,minimum,countsubl;
	scanf("%d",&n);
	int **cost=(int**)malloc((n+1)*sizeof(int*));
	for(i=0;i<=n;i++)
		cost[i]=(int*)malloc((n+1)*sizeof(int));
	int *dist=(int*)malloc((n+1)*sizeof(int));
	int *flag=(int*)malloc((n+1)*sizeof(int));
	for(i=1;i<=n;i++)
	{
		for(j=1;j<=n;j++)
		{
			scanf("%d",&cost[i][j]);
			if(cost[i][j]==0)
				cost[i][j]=infinity;
		}
	}
	for(i=1;i<=4;i++)
	{
		for(j=1;j<=n;j++)
			printf("%d ",cost[i][j] );
		printf("\n");
	}
	scanf("%d",&v);

	for(i=1;i<=n;i++)
	{
		flag[i]=0;
		dist[i]=cost[v][i];
		printf("%d\n",cost[v][i] );
	}
	flag[v]=1;
	count=2;
	while(count<=n)
	{
		minimum=infinity;
		for(w=1;w<=n;w++)
		{
			if(dist[w]<minimum&&flag[w]==0)
			{
				minimum=dist[w];
				u=w;
			}
		}
		flag[u]=1;
		for(w=1;w<=n;w++)
		{
			if(((dist[u]+cost[u][w])<dist[w])&&flag[w]==0)
				dist[w]=dist[u]+cost[u][w];
		}
		count++;
	}
	printf("\n Shortest Path: \n");
	for(i=1;i<=n;i++)
	{
		if(i!=v)
			printf("%d->%d, cost=%d\n",v,i,dist[i]);
	}
	return 0;
}

File: euler/other_imp/dijkstra.py
infinity=10**8
n=int(raw_input())
cost=[[0 for i in xrange(n+1)] for j in xrange(n+1)]
flag=[False]*(n+1)
dist=[0]*(n+1)
for i in xrange(1,n+1):
	arr=map(int,raw_input().strip().split())
	for j in xrange(1,n+1):
		if(arr[j-1]!=0): cost[i][j]=arr[j-1]
		else: cost[i][j]=infinity
# the starting vertex
v=int(raw_input())

flag[v]=True
for i in xrange(1,n+1):
	dist[i]=cost[v][i]
count=2
print 
while count<=n:
	u=1
	minimum=infinity
	for i in xrange(1,n+1):
		if dist[i]<minimum and flag[i]==False:
			minimum=dist[i]
			u=i

	flag[u]=True
	for i in xrange(1,n+1):
		if flag[i]==False and (dist[u]+cost[u][i])<dist[i]:
			dist[i]=dist[u]+cost[u][i]
	count+=1

print "The shortest paths are :"
for i in xrange(1,n+1):
	if i!=v:
		print v,"->",i,"cost=",dist[i]
